### PR TITLE
lib/rand: Break out random functions into separate package

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

The intention for this package is to provide a combination of the security of crypto/rand and the convenience of math/rand. It should be the first choice of random data unless ultimate performance is required and the usage is provably irrelevant from a security standpoint.

### Testing

Not changed.

### CODE REVIEW

File: lib/rand/random.go [hunk 1]
```diff
@@ -26,30 +31,36 @@ var (
 	defaultSecureRand = mathRand.New(defaltSecureSource)
 )
 
-func init() {
-	// The default RNG should be seeded with something good.
-	mathRand.Seed(RandomInt64())
-}
```
Comment (AudriusButkevicius) [inline lib/rand/random.go]: I guess cryptorand doesn't need seeding?

Comment (calmh) [inline lib/rand/random.go]: Indeed. I debated removing this, but we still use math/rand in a few places, and having it seeded by default somewhere is better than not doing it. We should do a cleanup later to not use math/rand anywhere, unless we need the predictable seeding or can show that we need the better performance of that package. And I don't think we ever need the performance as we're not generating that much random things

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as e6b78e5d56b1a426d08506c85075afede88e15fa. Thanks, @calmh!

**Pull request closed without merging**